node_io/lib/node_io.dart

```diff
@@ -72,11 +72,9 @@ set exitCode(int value) {
 ///
 /// On Windows the exit code can be set to any 32-bit value. However some of
 /// these values are reserved for reporting system errors like crashes.
-void exit([int code]) {
-  if (code is! int) {
-    throw ArgumentError('Integer value for exit code expected');
-  }
+Never exit(int code) {
```

Comment:
Is there a reason for this behavior change?

Reply:
I'm trying to match the API for `dart:io`, which has `exit` take in a non-nullable integer and return the new `Never` type.

Reply:
I see. Worth documenting in the changelog?

Reply:
Done